src/Admin/Polyfills.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Handles backward compatibility of assets for older versions of WP.
+ *
+ * @since 2.0
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Scripts;
+use WP_Styles;
+
+/**
+ * Registers assets that may not be available in the current site's version of core.
+ *
+ * @since 2.0
+ */
+final class Polyfills implements Delayed, Service, Registerable {
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'amp_register_polyfills';
```

Comment:
Why is this not using `wp_default_scripts` and `wp_default_styles`? 

Granted this would need to be not using `Delayed`, as the `register` method.

Reply:
I suppose not, though, as the intention is for the polyfills to only be run on specific screens.

Reply:
And `amp_register_default_scripts()` was doing this before?

Reply:
Yes, `amp_register_default_scripts` was doing that before. With the new approach it will only happen when `do-action( 'amp_register_polyfills' )` is called, and this has been placed throughout the plugin where needed.